zebrad/src/commands/start/sync.rs

```diff
@@ -149,23 +222,37 @@ where
                             }
                         }
                     }
-                    self.update_metrics();
                 }
+                self.update_metrics();
 
-                // If we have too many pending tasks, wait for one to finish:
-                if self.pending_blocks.len() > LOOKAHEAD_LIMIT {
-                    tracing::debug!(
-                        tips.len = self.prospective_tips.len(),
-                        pending.len = self.pending_blocks.len(),
-                        pending.limit = LOOKAHEAD_LIMIT,
-                        "waiting for pending blocks",
-                    );
+                // If we have too many pending tasks, wait for some to finish.
+                //
+                // Starting to wait is interesting, but logging each wait can be
+                // very verbose.
+                let mut first_wait = true;
+                while self.pending_blocks.len() > LOOKAHEAD_LIMIT {
+                    if first_wait {
+                        tracing::info!(
+                            tips.len = self.prospective_tips.len(),
+                            pending.len = self.pending_blocks.len(),
+                            pending.limit = LOOKAHEAD_LIMIT,
+                            "waiting for pending blocks",
+                        );
```

Comment:
Can this just be hoisted out of the loop?

Reply:
Sure, we could check the loop condition twice - once for the logging, and once for the loop.

It doesn't make sense to log "waiting for pending blocks" if we're not actually going to wait.

Reply:
Fixed in 53a2fcdc9e8f02bf3953014ba729a8937c75243c